### ISSUE TITLE
SqlServer IsClustered forced on identity

### ISSUE THREAD

**Issue by smint80** [dotnet/efcore#17770]
For SQL Server on EF Core 3.0:
``` C#
builder.HasKey(x => new { x.Provider, x.ProviderId });
builder.HasIndex(x => x.Id).IsUnique().IsClustered(false);
builder.Property(x => x.Id).IsRequired().UseIdentityColumn();
```
will result in failure aplying migration to SQL server as both the composite primary key and the unique identity will be created as clustered.
Using .HasAnnotation() to set SqlServer:IsClusted to false does not help.

Even though it is a good default to cluster by a identity column it should be possible to change behavier with IsClustered(false) as that will allow an identity column which is not the primary key.

### Further technical details

EF Core version: 3.0.0-rc2
Database provider: (e.g. Microsoft.EntityFrameworkCore.SqlServer)
Target framework: (e.g. .NET Core 3.0)

**Comment by roji** [comment]
Will take this.

**Comment by roji** [comment]
@smint80 I haven't been able to reproduce this. Please see the attempted repro below, which has a non-key identity-generated column with a unique index - the logs clearly show a unique index being created but without being clustered, and so no error is triggered. This works both in 3.0.0 (released) and 3.0.0-rc1.

```c#
class Program
{
    static void Main(string[] args)
    {
        using var ctx = new BlogContext();
        ctx.Database.EnsureDeleted();
        ctx.Database.EnsureCreated();
    }
}

public class BlogContext : DbContext
{
    public DbSet<Blog> Blogs { get; set; }

    static ILoggerFactory ContextLoggerFactory
        => LoggerFactory.Create(b => b.AddConsole().AddFilter("", LogLevel.Debug));

    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        => optionsBuilder
            .UseSqlServer(@"Server=localhost;Database=test;User=SA;Password=Abcd5678;Connect Timeout=60;ConnectRetryCount=0")
            .EnableSensitiveDataLogging()
            .UseLoggerFactory(ContextLoggerFactory);

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder.Entity<Blog>(blog =>
        {
            blog.HasKey(x => new { x.Id1, x.Id2 });
            blog.HasIndex(x => x.UniqueIdentityWithIndex).IsUnique(); //.IsClustered(false);
            blog.Property(x => x.UniqueIdentityWithIndex).IsRequired().UseIdentityColumn();
        });
    }
}

public class Blog
{
    public string Id1 { get; set; }
    public string Id2 { get; set; }
    public int UniqueIdentityWithIndex { get; set; }
}
```

**Comment by ajcvickers** [comment]
**EF Team Triage:** Closing this issue as the requested additional details have not been provided and we have been unable to reproduce it.

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*
